SignalClient.podspec

```diff
@@ -34,12 +34,18 @@ Pod::Spec.new do |s|
       'LIBSIGNAL_FFI_LIB_IF_NEEDED' => '$(PODS_TARGET_SRCROOT)/target/$(CARGO_BUILD_TARGET)/release/libsignal_ffi.a',
       'OTHER_LDFLAGS' => '$(LIBSIGNAL_FFI_LIB_IF_NEEDED)',
 
-      # Some day this will have to be updated for arm64 Macs (and the corresponding arm64 iOS simulator)
-      'CARGO_BUILD_TARGET[sdk=iphonesimulator*]' => 'x86_64-apple-ios',
+      'CARGO_BUILD_TARGET[sdk=iphonesimulator*][arch=arm64]' => 'aarch64-apple-ios-sim',
+      'CARGO_BUILD_TARGET[sdk=iphonesimulator*][arch=*]' => 'x86_64-apple-ios',
       'CARGO_BUILD_TARGET[sdk=iphoneos*]' => 'aarch64-apple-ios',
-      'CARGO_BUILD_TARGET[sdk=macosx*]' => 'x86_64-apple-darwin',
-      'ARCHS[sdk=iphonesimulator*]' => 'x86_64',
-      'ARCHS[sdk=iphoneos*]' => 'arm64',
```

Comment:
Looks like these were here to keep `pod lib lint` from trying to build for 32-bit platforms. CocoaPods/CocoaPods#5854 suggests we could try

```
'ARCHS[sdk=iphonesimulator*]' => '$(ARCHS_STANDARD_64_BIT)'
```

though that's from a few years ago, so it might need tweaking for Xcode 12.x.